sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Verifier.cs

```diff
@@ -0,0 +1,679 @@
+                {
+                    var codeActionsForDiagnostic = GetCodeActionsForDiagnostic(codeFixProvider, currentDocument, diagnostics[diagnosticIndexToFix]);
+
+                    if (TryGetCodeActionToApply(codeFixTitle, codeActionsForDiagnostic, out var codeActionToExecute))
+                    {
+                        currentDocument = ApplyCodeFix(currentDocument, codeActionToExecute);
+                        CalculateDiagnosticsAndCode(diagnosticAnalyzer, currentDocument, parseOption, out diagnostics, out actualCode);
+
+                        codeFixExecutedAtLeastOnce = true;
+                        codeFixExecuted = true;
+                    }
+                }
+            } while (codeBeforeFix != actualCode);
+
+            codeFixExecutedAtLeastOnce.Should().BeTrue();
+
+            var actualWithUnixLineEnding = actualCode.Replace(WindowsLineEnding, UnixLineEnding);
+            var expectedWithUnixLineEnding = File.ReadAllText(pathToExpected).Replace(WindowsLineEnding, UnixLineEnding);
+            actualWithUnixLineEnding.Should().Be(expectedWithUnixLineEnding);
+        }
+
+        private static void RunFixAllProvider(DiagnosticAnalyzer diagnosticAnalyzer, CodeFixProvider codeFixProvider,
+            string codeFixTitle, FixAllProvider fixAllProvider, Document document, ParseOptions parseOption, string pathToExpected)
+        {
+            var currentDocument = document;
+            CalculateDiagnosticsAndCode(diagnosticAnalyzer, currentDocument, parseOption, out var diagnostics, out var actualCode);
+
+            diagnostics.Should().NotBeEmpty();
+
+            var fixAllDiagnosticProvider = new FixAllDiagnosticProvider(
+                codeFixProvider.FixableDiagnosticIds.ToImmutableHashSet(),
+                (doc, ids, ct) => Task.FromResult(
+                    GetDiagnostics(currentDocument.Project.GetCompilationAsync(ct).Result, diagnosticAnalyzer)),
+                null);
+            var fixAllContext = new FixAllContext(currentDocument, codeFixProvider, FixAllScope.Document,
+                codeFixTitle,
+                codeFixProvider.FixableDiagnosticIds,
+                fixAllDiagnosticProvider,
+                CancellationToken.None);
+            var codeActionToExecute = fixAllProvider.GetFixAsync(fixAllContext).Result;
+
+            codeActionToExecute.Should().NotBeNull();
+
+            currentDocument = ApplyCodeFix(currentDocument, codeActionToExecute);
+
+            CalculateDiagnosticsAndCode(diagnosticAnalyzer, currentDocument, parseOption, out diagnostics, out actualCode);
+
+            var actualWithUnixLineEnding = actualCode.Replace(WindowsLineEnding, UnixLineEnding);
+            var expectedWithUnixLineEnding = File.ReadAllText(pathToExpected).Replace(WindowsLineEnding, UnixLineEnding);
+            actualWithUnixLineEnding.Should().Be(expectedWithUnixLineEnding);
```

Comment:
Same here I force unix line ending on both actual and expected. Note that we still need to do it on the `actualCode` because the fix can provide other line returns (for example header fix).

Reply:
I can't decide whether it would help if there was a _FormatAsUnix_ extension method. It might make the code slightly easier to read, but it's marginal so probably not worth changing.